src/BlogContentSection.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Col, Row } from 'react-bootstrap'
+import request from 'superagent'
+import moment from 'moment'
+import smartcrop from 'smartcrop'
+
+import PageSection from './PageSection'
+
+
+const getImgSrc = (str) => {
+    const start = str.indexOf('src="') + 5
+    const end = str.substring(start).indexOf('"') + start
+    return str.substring(start, end)
+}
+
+const formatDate = (dateString) => {
+    return moment(dateString, 'YYYY-MM-DD HH:mm:ss').format('MM/DD/YYYY')
+}
+
+const htmlDecode = (input) => {
+    const doc = new DOMParser().parseFromString(input, 'text/html')
+    return doc.documentElement.textContent
+}
+
+const CANVAS_WIDTH = 600
+
+const CANVAS_HEIGHT = 400
+
+const loadImage = (src) => {
+    return new Promise((resolve) => {
+        let img = new Image()
+        img.onload = () => {
+            resolve(img)
+        }
+        img.crossOrigin = 'Anonymous'
+        img.src = src
+    })
+}
+
+class ImageCrop extends Component {
+
+    componentDidMount() {
+        const { src } = this.props
+        const ctx = this.refs.canvas.getContext('2d')
+        let img = null
+
+        loadImage(src)
+            .then((loadedImg) => {
+                img = loadedImg
+                smartcrop.crop(img, {
+                    width: CANVAS_WIDTH,
+                    height: CANVAS_HEIGHT,
+                    minScale: 1,
+                    ruleOfThirds: true
+                }).then((result) => {
+                    const crop = result.topCrop
+                    ctx.drawImage(img, crop.x, crop.y, crop.width, crop.height,
+                        0, 0, CANVAS_WIDTH, CANVAS_HEIGHT)
+                })
+            })
+    }
+
+    render() {
+        return <canvas ref="canvas"
+            width={CANVAS_WIDTH} height={CANVAS_HEIGHT}></canvas>
+    }
+}
+
+ImageCrop.propTypes = {
+    src: PropTypes.string
+}
+
+class BlogContentSection extends Component {
+    constructor(props) {
+        super(props)
+
+        this.state = {
+            blogItems: []
+        }
+    }
+
+    componentDidMount() {
+        request
+            .get('https://api.rss2json.com/v1/api.json')
```

Comment:
I want to replace this with our own backend desperately. What's the worst case if they return malicious data? If it's anything but an ugly page let's see about self hosting or putting up our own endpoint... Or even doing it entirely client-side

Reply:
Good call out. I'm not sure what security measures https://rss2json.com/docs takes, if any, if malicious data is sent. I couldn't find any information, and I don't think the code is open source - I checked the developer's github and there's no repo for this code. Sorry for not thinking about security!

We can do any of the options you mentioned, but what do you want to do in the mean time? I can either remove the blog content section or just hard code this content temporarily until we have the endpoint/self-hosting ready.

Reply:
Let's hard code it in the meantime